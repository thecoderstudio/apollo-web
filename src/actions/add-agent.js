const SELECT_OPERATING_SYSTEM = 'SELECT_OPERATING_SYSTEM';
const SELECT_ARCHITECTURE = 'SELECT_ARCHITECTURE';

function selectOperatingSystem(operatingSystem) {
  return {
    type: SELECT_OPERATING_SYSTEM,
    selectedOperatingSystem: operatingSystem
  };
}

function selectArchitecture(architecture) {
  return {
    type: SELECT_ARCHITECTURE,
    selectedArchitecture: architecture
  };
}

export {
  selectOperatingSystem,
  SELECT_OPERATING_SYSTEM,
  selectArchitecture,
  SELECT_ARCHITECTURE
};
