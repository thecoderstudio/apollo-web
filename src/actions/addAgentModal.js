const SHOW_ADD_AGENT_MODAL = 'SHOW_ADD_AGENT_MODAL';
const CLOSE_ADD_AGENT_MODAL = 'CLOSE_ADD_AGENT_MODAL';

function showAddAgentModal() {
  return {
    type: SHOW_ADD_AGENT_MODAL
  };
};

function closeAddAgentModal() {
  return {
    type: SHOW_ADD_AGENT_MODAL
  };
};

export { showAddAgentModal, SHOW_ADD_AGENT_MODAL, closeAddAgentModal, CLOSE_ADD_AGENT_MODAL};
