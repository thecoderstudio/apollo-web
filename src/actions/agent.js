export const LIST_AGENTS = 'LIST_AGENTS';

export const listAgents = (agents) => ({
  type: LIST_AGENTS,
  agents
});