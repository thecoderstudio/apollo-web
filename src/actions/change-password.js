export const PROMPTED_PASSWORD_CHANGE = 'PROMPTED_PASSWORD_CHANGE';

export function promptedPasswordChange() {
  return {
    type: PROMPTED_PASSWORD_CHANGE
  };
}
