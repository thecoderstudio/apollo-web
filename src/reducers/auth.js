import Cookies from 'js-cookie';

function logout() {
  Cookies.remove('AIOHTTP_SESSION', {path: '/'});
}

export default function authReducer(state = { authenticated: false }, action) {
  if (action.type == 'LOGOUT') {
    logout();
  }
  return { authenticated: Boolean(Cookies.get('AIOHTTP_SESSION', { path: '/' })) };
}
