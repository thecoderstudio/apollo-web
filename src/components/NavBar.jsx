import React from 'react';
import styled from 'styled-components';
import { logout as logoutAction } from '../actions/auth';
import Text from './Text';

const NavigationBar = styled.div`
  height: 50px;
  padding: 10px;
  display: grid;
  grid-template-columns: 1fr [options] 300px;

  background-color: ${props => props.theme.lightBlack};
`;

const Logout = styled(Text)`
  float: right;
  max-width: 300px;

  &:hover {
    cursor: pointer;
  }
`;

class NavBar extends React.PureComponent {
  constructor(props) {
    super(props);
    this.logout = this.logout.bind(this);
  }

  logout() {
    let { dispatch } = this.props;
    dispatch(logoutAction());
  }

  render() {
    return (
      <NavigationBar>
        <Logout onClick={this.toggleDropDown}>Logout</Logout>
      </NavigationBar>
    );
  }
}

export default  (NavBar);