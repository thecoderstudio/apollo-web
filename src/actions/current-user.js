export const CACHE_CURRENT_USER = 'CACHE';
export const REMOVE_CURRENT_USER = 'REMOVE';
import { parseSnakeCaseObj } from '../util/parser';

function removeCurrentUser() {
  return {
    type: REMOVE_CURRENT_USER
  };
}

function cacheCurrentUser(user) {
  return {
    type: CACHE_CURRENT_USER,
    user: parseSnakeCaseObj(user)
  };
}

export { removeCurrentUser, cacheCurrentUser };
