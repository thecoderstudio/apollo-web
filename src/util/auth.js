import { store } from '../store';
import { logout } from '../actions/auth';
import { removeCurrentUser } from '../actions/current-user';

export function logout() {
  store.dispatch(logout());
  store.dispatch(removeCurrentUser());
}
