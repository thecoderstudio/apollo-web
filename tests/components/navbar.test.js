import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import waitForExpect from 'wait-for-expect';
import NavBar from '../../src/components/NavBar';
import { logout as logoutAction } from '../../src/actions/auth';

const mockStore = configureStore([]);

function getComponent(store) {
  return renderer.create(
    <Provider store={store}>
      <NavBar />
    </Provider>
  );
}

describe('login', () => {
  let store;
  let spy;

  beforeEach(() => {
    store = mockStore({});
    spy = jest.spyOn(store, 'dispatch');
  });

  it("renders correctly", () => {
    let tree = getComponent(store).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("handles successful logout", async () => {
    let tree = getComponent(store);
    const instance = tree.root
    instance.findByType('button').props.onClick()
    expect(tree).toMatchSnapshot()
  });
});
