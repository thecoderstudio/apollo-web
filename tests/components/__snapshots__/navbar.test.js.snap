// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login handles successful logout 1`] = `
<div
  className="sc-AxiKw cQhYjZ"
>
  <button
    className="sc-AxjAm sc-AxirZ sc-AxhCb egA-DhE"
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;

exports[`login renders correctly 1`] = `
<div
  className="sc-AxiKw cQhYjZ"
>
  <button
    className="sc-AxjAm sc-AxirZ sc-AxhCb egA-DhE"
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;
